python/tests/test_aac/plugins/no_ext_for_final/test_no_ext_for_final.py

```diff
@@ -1,38 +1,55 @@
-from typing import Tuple
-from click.testing import CliRunner
-from aac.execute.command_line import cli, initialize_cli
-from aac.execute.aac_execution_result import ExecutionStatus
 from aac.context.language_context import LanguageContext
-
-
-from aac.plugins.no_ext_for_final.no_ext_for_final_impl import plugin_name
-
-
 from aac.plugins.no_ext_for_final.no_ext_for_final_impl import no_extension_for_final
 
 
 class TestNoExtForFinal(TestCase):
-    def test_no_extension_for_final(self):
+
+    #  Nominal (expected pass) test with a single valid Schema definition
+    def test_no_extension_for_final_nominal(self):
         context = LanguageContext()
         schema_definition = context.get_definitions_by_name("aac.lang.Schema")
         if len(schema_definition) != 1:
             self.fail("Expected to find one and only one Schema definition")
         schema_definition = schema_definition[0]
+
         definitions = context.parse_and_load(root_schema_with_good_ext)
         result = no_extension_for_final(definitions[0].instance, definitions[0], schema_definition.instance)
         self.assertTrue(result.is_success())
         result = no_extension_for_final(definitions[1].instance, definitions[1], schema_definition.instance)
         self.assertTrue(result.is_success())
         context.remove_definitions(definitions)
 
-        definitions = context.parse_and_load(root_schema_with_bad_ext)
+
+    #  Off-nominal (expected fail) tests
+    def test_no_extension_for_final_offnominal(self):
+        context = LanguageContext()
+        schema_definition = context.get_definitions_by_name("aac.lang.Schema")
+        if len(schema_definition) != 1:
+            self.fail("Expected to find one and only one Schema definition")
+        schema_definition = schema_definition[0]
+
+
+        #  Off-nominal (expected fail) test with a circular schema definition
+        #  TestSchema tries to extend TestSchema
+        definitions = context.parse_and_load(root_schema_with_bad_ext_circular)
         result = no_extension_for_final(definitions[0].instance, definitions[0], schema_definition.instance)
         self.assertTrue(result.is_success())
         result = no_extension_for_final(definitions[1].instance, definitions[1], schema_definition.instance)
         self.assertFalse(result.is_success())
         context.remove_definitions(definitions)
 
+
+        #  Off-nominal (expected fail) tests with a invalid schema definition
+        #  TestSchema defined twice
+        definitions = context.parse_and_load(root_schema_with_bad_ext_duplicate)
+        result = no_extension_for_final(definitions[0].instance, definitions[0], schema_definition.instance)
+        self.assertTrue(result.is_success())
+        result = no_extension_for_final(definitions[1].instance, definitions[1], schema_definition.instance)
+        self.assertFalse(result.is_success())
+        #  context.remove_definitions(definitions) #  Intentionally commented out
```

Comment:
So is this actually needing to remain? If so, need more explanation for why its staying but commented out. If not, needs to be removed.

Reply:
Removed.

Reply:
Looks like the does not exist test had a similar line commented out, so that one now needs removed.